File: numass-control/cryotemp/src/main/java/inr/numass/control/cryotemp/PKT8DeviceFactory.java
package inr.numass.control.cryotemp;

import hep.dataforge.context.Context;
import hep.dataforge.control.devices.Device;
import hep.dataforge.meta.Meta;
import inr.numass.control.DeviceViewConnection;
import inr.numass.control.DeviceViewFactory;

/**
 * Created by darksnake on 09-May-17.
 */
public class PKT8DeviceFactory implements DeviceViewFactory {
    @Override
    public String getType() {
        return PKT8Device.PKT8_DEVICE_TYPE;
    }

    @Override
    public PKT8Device build(Context context, Meta meta) {
        return new PKT8Device(context, meta);
    }

    @Override
    public DeviceViewConnection buildView(Device device) {
        return PKT8View.build(device.getContext());
    }
}

File: numass-control/cryotemp/src/main/kotlin/inr/numass/control/cryotemp/PKT8DeviceFactory.kt
package inr.numass.control.cryotemp

import hep.dataforge.context.Context
import hep.dataforge.control.devices.DeviceFactory
import hep.dataforge.meta.Meta

/**
 * Created by darksnake on 09-May-17.
 */
class PKT8DeviceFactory : DeviceFactory {
    override val type: String = PKT8Device.PKT8_DEVICE_TYPE

    override fun build(context: Context, meta: Meta): PKT8Device {
        return PKT8Device(context, meta)
    }
}
